DPL GUI: add GUI to enable / disable driver signposts

ktf opened the pull request:
DPL GUI: add GUI to enable / disable driver signposts

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/AliceO2Group/AliceO2/pull/12533).
* __->__ #12533
* #12529 (5 commits)

alibuild commented:
Error while checking build/O2/fullCI for b88a52e5631f03c19fb49503de5e91d8256bd587 at 2024-01-16 15:47:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
command /sw/slc8_x86-64/O2/12533-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh had nonzero exit code 128
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 0 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/52443 was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data HMP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FLP/DISTSUBTIMEFRAME/0 was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MID/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data CTP/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TOF/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TPC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data ITS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MFT/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data MCH/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data PHS/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data TRD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data CPV/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data EMC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FDD/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FT0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data FV0/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73540:raw-file-reader]: [14:47:23][ERROR] Expected Lifetime::Timeframe data ZDC/RAWDATA was not created for timeslice 1 and might result in dropped timeframes
[73565:BadMapCalibSpec]: [14:47:42][ERROR] Insufficient statistics: 5 entries in lowE histo, do nothing
[73689:ctf-writer]: [14:47:46][ERROR] Some Lifetime::Timeframe data got dropped starting at 1
[ERROR] Workflow crashed - PID 73661 (tpc-calibrator-gainmap-tracks) did not exit correctly however it's not clear why. Exit code forced to 128.
[ERROR]  - Device tpc-calibrator-gainmap-tracks: pid 73661 (exit 128)
[ERROR] SEVERE: Device tpc-calibrator-gainmap-tracks (73661) returned with 128
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12533/b88a52e5631f03c19fb49503de5e91d8256bd587/build_O2_fullCI/pretty.html).

ktf committed:
DPL use Signposts to debug consumeWhenAll

ktf committed:
DPL: drop unused o2_signpost_id_make_with_pointer method

Probably old remaining from a previous implementation.

ktf committed:
DPL: rework signposts support

The logger based Signposts are now behaving the same on macOS and linux when
running normally.

The difference between the two implementations will now be that if Instruments
enables some OSLog namespace, we stop printing the messages and use the os_signpost
API instead.

ktf committed:
DPL GUI: add GUI to enable / disable driver signposts

ktf committed:
DPL: add proper tracing for DataProcessingDevice socket callbacks

ktf committed:
Ability to turn on / off the logger from the GUI.
